include/pybind11/pytypes.h

```diff
@@ -890,6 +895,22 @@ class iterator : public object {
     object value = {};
 };
 
+
+
+class type : public object {
+public:
+    PYBIND11_OBJECT_CVT(type, object, PyType_Check, detail::Py_TYPE_Convert);
```

Comment:
@bstaletic wasn't a fan of the implicit conversion from `object` to `type`, though?

Reply:
I could be convinced otherwise, but it does seem odd that this would be possible:

```c++
auto f(py::type) {}

int main() { py::object i = py::int_(3); f(i); }
```

Reply:
I think I agree here. I'd be happy with this as well, for example:
```c++
PYBIND11_OBJECT(type, object, PyType_Check)

explicit type(handle object) : type((PyObject*) Py_TYPE(obj), borrowed_t{}) {}
```